ruby/src/main/java/com/google/protobuf/jruby/RubyMap.java

```diff
@@ -189,8 +190,25 @@ public IRubyObject index(ThreadContext context, IRubyObject key) {
      */
     @JRubyMethod(name = "==")
     public IRubyObject eq(ThreadContext context, IRubyObject _other) {
-        if (_other instanceof RubyHash)
-            return toHash(context).op_equal(context, _other);
+        if (_other instanceof RubyHash){
+            // quick check to see if they are equivalent
+            IRubyObject matched = toHash(context).op_equal(context, _other);
+            if (matched == context.runtime.getTrue()) {
+                return context.runtime.getTrue();
+            }
+
+            RubyHash other = (RubyHash)_other;
```

Comment:
Hmm -- I assume you actually hit the test failure then wrote this to fix it? I was under the impression that a Ruby hash would do a deep equality comparison already (using the `:==` method on values as appropriate) -- the [docs](http://ruby-doc.org/core-2.2.0/Hash.html#method-i-3D-3D) seem to say this too. Perhaps JRuby behaves differently?